pkgs/development/libraries/xgboost/default.nix

```diff
@@ -77,11 +85,7 @@ stdenv.mkDerivation rec {
     # https://github.com/dmlc/xgboost/blob/ccf43d4ba0a94e2f0a3cc5a526197539ae46f410/CMakeLists.txt#L145
     "-DCMAKE_C_COMPILER=${cudaPackages.backendStdenv.cc}/bin/gcc"
     "-DCMAKE_CXX_COMPILER=${cudaPackages.backendStdenv.cc}/bin/g++"
-  ] ++ lib.optionals
-    (cudaSupport
-      && lib.versionAtLeast cudaPackages.cudatoolkit.version "11.4.0")
-    [ "-DBUILD_WITH_CUDA_CUB=ON" ]
```

Comment:
Is this no longer required when building with an older CUDA?

Reply:
It looks like nothing prior to 11.x is supported: https://github.com/dmlc/xgboost/blob/1fdcd2b7dafc27e15a655f7c8be237ad4a502e76/CMakeLists.txt#L162-L165

Additionally, anything prior to 11.4 fails to build (maybe because of the requirement for a C++17 compiler?):

- 11.0: https://gist.github.com/ConnorBaker/2f8fc7f64c2ade1fe699bfaa7a314e34
- 11.1: https://gist.github.com/ConnorBaker/d5faef2121af779d6564eb476268273a
- 11.2: https://gist.github.com/ConnorBaker/be3051f67ecfe17ffe31c312c9b50a87
- 11.3: https://gist.github.com/ConnorBaker/c042220a7fe45409206b9ee88cbaec39

These were all done with that flag added back.

Reply:
Added https://github.com/NixOS/nixpkgs/pull/314711/commits/d484d98d9334efa4abefd4d9e4110c266949c29f -- thoughts?

Reply:
LGTM